packages/gql-utils/data-loader/create-data-loader.ts

```diff
@@ -0,0 +1,14 @@
+import DataLoader, { type BatchLoadFn } from "dataloader";
+
+export function createDataLoader<TKey, TValue>(
+  batchFn: BatchLoadFn<TKey, TValue>,
+  onComplete?: (values: TValue[]) => void
+) {
+  return new DataLoader(async (keys: readonly TKey[]) => {
+    const result = (await batchFn(keys)) ?? [];
+
+    onComplete?.(result as unknown as TValue[]);
```

Comment:
@sevilayerkan let's add a test for this loader and cover this case, we can use as an example

https://github.com/kamp-us/monorepo/blob/566babc6fdacbec072ba865cff36e5c4d680060b/packages/gql-utils/prisma/create-loader.test.ts#L33-L45